### PR TITLE
Fixed bugs with menu

### LINKED ISSUE #1214: "Save Settings to Disk" no longer available
#### Software and hardware versions 
Collect v1.8.0; Galaxy Tab A, Android 5.1.1 and Moto G4, Android 7.0.0

#### Problem description
Regression of #1138; the "Save Settings to Disk" menu item is no longer available from Admin Settings.

#### Steps to reproduce the problem
Go to the Admin Settings.

#### Expected behavior
"Save Settings to Disk" is available from the three dots menu as in previous versions.

### PR DESCRIPTION
**Pull request by grzesiek2010**
In reference to resolve #1214 & #1211

Ok so I removed all checks for device menu button eg https://github.com/opendatakit/collect/pull/1218/files#diff-aed1c0d39875468b38ba2a5f678fd46bL95 
I think it's fishy and it causes the problems.

It was a bad decision to implement it. 
Now action menu will be always visible even if your device has the menu button, it's a default android behavior when you use support library so we shouldn't force hiding the menu.

@shobhitagarwal1612 thanks for discussion

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java [hunk 1]
```diff
@@ -551,64 +539,8 @@ public boolean accept(File pathname) {
...
-                        .getFormController();
-                Collect.getInstance()
-                        .getActivityLogger()
-                        .logInstanceAction(this, "onOptionsItemSelected",
-                                "MENU_HIERARCHY_VIEW");
-                if (formController.currentPromptIsQuestion()) {
-                    saveAnswersForCurrentScreen(DO_NOT_EVALUATE_CONSTRAINTS);
-                }
-
-                timerLogger.logTimerEvent(TimerLogger.EventTypes.HIERARCHY, 0, null, false, true);
```
Comment (shobhitagarwal1612) [inline collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:589]: Is it necessary to remove this line?

Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:589]: I just remove duplicated code, it's here https://github.com/grzesiek2010/collect/blob/a2e4192de81cd0df20fccb913c1a71806bc6ae0a/collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java#L936 the same line

**Comment by mmarciniak90** [comment]
Tested this fix with success.
Verified on Android 4.1, 4.2, 4.4, 5.1, 6.0
Menu icon (3 dots) is now visible even if device has a menu button.
Additionally, share icon is visible. It was a bug.
Setting can be saved on device without menu button. It was not possible before fix
screenshot_2017-06-28-15-55-49
screenshot_2017-06-28-15-56-13

**Comment by grzesiek2010** [comment]
@mmarciniak90 thanks for testing. Looks like we are only waiting for a feedback https://forum.opendatakit.org/t/odk-collect-v1-8-0-hid-settings-menu/7065/9

**Comment by lognaturel** [comment]
Only showing the three dots in the absence of a hardware button was a good concept but ultimately you're right that it's best not to fight defaults. We live and learn!

I've done a little bit of additional checking to verify that language selection, form list search/sort weren't affected. So far everything looks great. I have to be at the convening today but will do a last pass and merge at around 5:45pm PST. Then we should be ready for a 1.8.1 release.

Thanks and go team! 🚀

File: collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java (hunk 1 above)
Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java:589]: :+1:

**Pull request merged**